torchrl/envs/transforms/transforms.py

```diff
@@ -4638,15 +4638,20 @@ def transform_observation_spec(
             raise ValueError(
                 f"observation_spec was expected to be of type CompositeSpec. Got {type(observation_spec)} instead."
             )
-        for key, spec in self.primers.items():
-            if spec.shape[: len(observation_spec.shape)] != observation_spec.shape:
-                expanded_spec = self._expand_shape(spec)
-                spec = expanded_spec
+        try:
+            device = observation_spec.device
```

Comment:
when is observation_spec.device returning an `RuntimeError`?
IIRC this should not happen anymore